ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/PermissionChecker.java

```diff
@@ -94,15 +94,10 @@ private boolean isChangingRestrictedSettings(RoleOptions options)
 
     private boolean hasPermissionToAlterRole(AuthenticatedUser performer, RoleResource roleResource)
     {
-        for (IResource resource : Resources.chain(roleResource))
-        {
-            Set<Permission> grantedPermissions = getPermissions(performer, resource);
-            if (grantedPermissions.contains(Permission.ALTER))
-            {
-                return true;
-            }
-        }
-        return false;
+        return Resources.chain(roleResource)
+                        .stream()
+                        .map(resource -> getPermissions(performer, resource))
```

Comment:
Again, not necessarily cleaner/better code, but an alternative:

```
.map(resource -> getPermission(performer, resource))
.flatMap(Set::stream)
.anyMatch(Permission.ALTER::equals);
```

Reply:
Looks better IMO. Question, would this have any measurable impact on perfomrance/heap preassure?